### PR TITLE
Add StackMode::raw for unsymbolized addresses

### PR DESCRIPTION
**Pull request by jordalgo**
Summary:
Some tools that use bpftrace would rather do their own symbolication (e.g. with llvm or a symbolication service). Using 'raw' as a StackMode outputs just the raw addresses without bpftrace doing the symbolication.

Test Plan:
Updated unit tests and tested locally:
```
sudo ./src/bpftrace -e 'kprobe:do_nanosleep { printf("Jordan\n%s\n",
ustack(raw));  }'
```
Example output:
```
Attaching 1 probe...
Jordan

7f058de12155

Jordan

7f088cae85b3
7f088caed253
1f3418b
6187e12
7f088cedf2e5
7f088ca979bf
7f088cb288ac
```

Reviewers:

Subscribers:

Tasks:

Tags:

##### Checklist

- [x] Language changes are updated in `man/adoc/bpftrace.adoc` and if needed in `docs/reference_guide.md`
- [x] User-visible and non-trivial changes updated in `CHANGELOG.md`
- [x] The new behaviour is covered by tests

### CODE REVIEW

File: docs/reference_guide.md
Comment (viktormalik) [inline docs/reference_guide.md:2957]: This change is irrelevant to this PR, I'd prefer not including it.

File: docs/reference_guide.md
Comment (viktormalik) [inline docs/reference_guide.md:3754]: Same as above.

**Review by viktormalik — COMMENTED** [review]
Seems like a useful feature, thanks!

Apart from the two nits below, could you also expand the tests? Ideally check where `[ku]stack(perf)` is used - runtime, codegen, semantic analyser.

**Comment by jordalgo** [comment]
@viktormalik Workflow question. Is force pushing changes ok or would you prefer commits stacked on top with changes and then possibly squashing after approval?

**Comment by viktormalik** [comment]
> @viktormalik Workflow question. Is force pushing changes ok or would you prefer commits stacked on top with changes and then possibly squashing after approval?

I generally prefer force-pushing, especially with small PRs like this one.

File: docs/reference_guide.md
Comment (viktormalik) [inline docs/reference_guide.md:2342]: What is this change?

File: docs/reference_guide.md
Comment (viktormalik) [inline docs/reference_guide.md:2957]: Same problem as before - please do not update lines unrelated to the PR.

File: tests/codegen/call_kstack.cpp
Comment (viktormalik) [inline tests/codegen/call_kstack.cpp:19]: I don't think that this is going to generate the same codegen as the previous two (three `kstack` calls vs two). Maybe rather add:
```cpp
test("kprobe:f { @x = kstack(raw); @y = kstack(raw, 6) }", result);
```

File: tests/codegen/call_ustack.cpp
Comment (viktormalik) [inline tests/codegen/call_ustack.cpp:19]: Same as above with `kstack`.

File: docs/reference_guide.md
Comment (jordalgo) [reply docs/reference_guide.md:2957]: My apologies. Still fixing some things...

**Comment by ajor** [comment]
Will external symbolisers be able to tell which process a raw stack trace comes from? It might be useful to print some sort of identifier along with the stack.

PID is the obvious one, but with the changes going on in #2386, maybe we'd want to also support other options like exe-path + base-offset.

**Comment by jordalgo** [comment]
@ajor Good thought. I think I had initially assumed that logging the pid or exe-path per stack would be the responsibility of whoever was writing the script. If we added these items to the top of each stack would we do it for "perf" and "bpftrace" modes as well just for consistency? I defer to the bpftrace experts on what would be the most logical.

**Comment by danobi** [comment]
GNU_BUILD_ID might be worth looking into. IIRC @olsajiri was involved in some recent kernel work involving that. But 
I think it encountered some resistance

**Comment by danobi** [comment]
I dont have strong opinion on who/what should print the extra context.

**Comment by ajor** [comment]
> logging the pid or exe-path per stack would be the responsibility of whoever was writing the script

I think this makes sense as a place to start actually. It'll be the most flexible and we could always introduce another stack type with these extra identifiers later.

File: tests/codegen/call_kstack.cpp
Comment (viktormalik) [inline tests/codegen/call_kstack.cpp:17]: Add the same into `call_uprobe.cpp`?

**Review by viktormalik — APPROVED** [review]
Agreed with @ajor, let's leave it up to the user to log the PID/exe for now.

LGTM, one nit below.

File: src/bpftrace.cpp
Comment (ajor) [inline src/bpftrace.cpp:1799]: Could we add a `LOG(BUG)` in here as a precaution against future programming errors?

**Review by ajor — APPROVED** [review]
Looks good, just one nit

**Comment by jordalgo** [comment]
Is there something I need to do to re-trigger the CI tests?

**Comment by jordalgo** [comment]
Should I also update the changelog for this?

**Comment by danobi** [comment]
> Should I also update the changelog for this?

Yes, please do

**Comment by jordalgo** [comment]
Added changelog item.

**Comment by viktormalik** [comment]
Ah, there's a merge conflict in the changelog, can you please rebase? Thanks!

**Pull request merged**
